src/it/add-third-party-missing-file/pom.xml

```diff
@@ -0,0 +1,110 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Lesser Public License for more details.
+
+  You should have received a copy of the GNU General Lesser Public
+  License along with this program.  If not, see
+  <http://www.gnu.org/licenses/lgpl-3.0.html>.
+  #L%
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.codehaus.mojo.license.test</groupId>
+  <artifactId>add-third-party-missing-file</artifactId>
+  <version>@pom.version@</version>
+
+  <name>License Test :: add-third-party-missing-file</name>
+
+  <packaging>jar</packaging>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <license.verbose>true</license.verbose>
+    <license.failOnMissing>true</license.failOnMissing>
+    <license.useMissingFile>true</license.useMissingFile>
+    <license.force>true</license.force>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.codehaus.jettison</groupId>
+      <artifactId>jettison</artifactId>
+      <version>1.1</version>
+    </dependency>
+    <dependency>
+      <groupId>org.json</groupId>
+      <artifactId>json</artifactId>
+      <version>20070829</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <pluginManagement>
+      <plugins>
+        <plugin>
+          <groupId>org.codehaus.mojo</groupId>
+          <artifactId>license-maven-plugin</artifactId>
+          <version>@pom.version@</version>
+          <dependencies>
+            <dependency>
+              <groupId>org.license</groupId>
+              <artifactId>license-repo</artifactId>
+              <version>1.0.0</version>
+              <scope>system</scope>
+              <systemPath>${basedir}/license-repo.jar</systemPath>
+            </dependency>
+          </dependencies>
+        </plugin>
+      </plugins>
+    </pluginManagement>
+
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>license-maven-plugin</artifactId>
+        <version>@pom.version@</version>
+        <executions>
+          <execution>
+            <id>by-classpath</id>
+            <phase>generate-resources</phase>
+            <goals>
+              <goal>add-third-party</goal>
+            </goals>
+            <configuration>
+              <missingFileUrl>classpath:missing-licenses.properties</missingFileUrl>
+            </configuration>
+          </execution>
+          <execution>
+            <id>by-file</id>
+            <phase>generate-resources</phase>
+            <goals>
+              <goal>add-third-party</goal>
+            </goals>
+            <configuration>
+              <missingFileUrl>file:///${basedir}/missing-licenses.properties</missingFileUrl>
```

Comment:
I think both executions write to the same report file. You probably need to configure separate files per execution.

Reply:
ok

Reply:
Funny that `file:///${basedir}` works. IIRC, ${basedir} tends to be an absolute path, which on Linux gives four slashes - e.g. `file:////path/to/basedir`. I thought, ${project.baseUri} is the only reliable way to test situations like this.